### ISSUE TITLE
Missing export items

### ISSUE THREAD

**Issue by BrutuZ** [ZeroQI/Lambda.bundle#4]
Having issues trying to export Episode NFO and Special Season's Poster and Fanart.
Below is the file structure after refreshing metadata.
```
\<show>\fanart.jpg
\<show>\poster.jpg
\<show>\theme.mp3
\<show>\tvshow.nfo
\<show>\Season 00
\<show>\Season 00\<show>s00e04.mp4
\<show>\Season 01
\<show>\Season 01\Season01-fanart.jpg
\<show>\Season 01\Season01.jpg
\<show>\Season 01\<show>s01e01.mp4
\<show>\Season 01\<show>s01e02.mp4
\<show>\Season 01\<show>s01e03.mp4
\<show>\Season 01\<show>s01e04.mp4
\<show>\Season 01\<show>s01e05.mp4
\<show>\Season 01\<show>s01e06.mp4
\<show>\Season 01\<show>s01e07.mp4
\<show>\Season 01\<show>s01e08.mp4
\<show>\Season 01\<show>s01e08.nfo
```
Both seasons have custom posters set on Plex, yet `season-specials-poster.jpg` is missing.
Only the show's last episode's NFO is being generated (the plugin expects each episode to be in its own folder?). This also happens with multiple regular seasons.

#### Suggestions:
- Only export if Season_Art != Show_Art to avoid multiple copies of the same image with different filenames. In the above example `Season01-fanart.jpg` and `fanart.jpg` are the same since only the show fanart is set
- Switch the preferences from lists to input so we can customize filenames.
---
Excerpt from the log showing ValidatePrefs():
```
======================================================================================================
ValidatePrefs() - PlexRoot: 'Plex\Data\Plex Media Server'
# agent settings json file: 'Plug-ins\Lambda.bundle\Contents\DefaultPrefs.json'
# agent settings xml prefs: 'Plug-in Support\Preferences\com.plexapp.agents.Lambda.xml'
[X] Prefs['series_themes'    ] = 'theme.mp3'                                     , default = 'Ignored'
[ ] Prefs['series_banner'    ] = 'Ignored'                                       , default = 'Ignored'
[X] Prefs['series_poster'    ] = 'poster.ext'                                    , default = '{}.ext'
[X] Prefs['series_fanart'    ] = 'fanart.ext'                                    , default = 'Ignored'
[X] Prefs['series_nfo'       ] = 'tvshow.nfo'                                    , default = 'Ignored'
[ ] Prefs['season_banner'    ] = 'Ignored'                                       , default = 'Ignored'
[X] Prefs['season_poster'    ] = 'season-specials-poster.ext|Season{}.ext'       , default = 'Ignored'
[X] Prefs['season_fanart'    ] = 'season-specials-fanart.ext|Season{}-fanart.ext', default = 'Ignored'
[ ] Prefs['episode_thumbs'   ] = 'Ignored'                                       , default = 'Ignored'
[X] Prefs['episode_nfo'      ] = '{}.nfo'                                        , default = 'Ignored'
[X] Prefs['movies_poster'    ] = '{}.ext'                                        , default = '{}.ext'
[X] Prefs['movies_fanart'    ] = '{}-fanart.ext'                                 , default = 'Ignored'
[X] Prefs['movies_nfo'       ] = '{}.nfo'                                        , default = 'Ignored'
[ ] Prefs['artist_poster'    ] = 'Ignored'                                       , default = 'Ignored'
[ ] Prefs['artist_fanart'    ] = 'Ignored'                                       , default = 'Ignored'
[ ] Prefs['artist_nfo'       ] = 'Ignored'                                       , default = 'Ignored'
[ ] Prefs['album_poster'     ] = 'Ignored'                                       , default = 'Ignored'
[ ] Prefs['album_fanart'     ] = 'Ignored'                                       , default = 'Ignored'
[ ] Prefs['album_nfo'        ] = 'Ignored'                                       , default = 'Ignored'
[ ] Prefs['collection_poster'] = 'Ignored'                                       , default = 'Ignored'
[ ] Prefs['collection_fanart'] = 'Ignored'                                       , default = 'Ignored'
[ ] Prefs['collection_resume'] = 'Ignored'                                       , default = 'Ignored'
[X] Prefs['collection_folder'] = 'root'                                          , default = 'root'
[X] Prefs['metadata_source'  ] = 'local'                                         , default = 'local'
[ ] Prefs['reset_to_defaults'] = 'False'                                         , default = 'false'
=======================================================================================================
```

**Comment by ZeroQI** [comment]
Episode NFOs bug confirmed, didn't code to save multiple in structure, was only testing single eps, didn't spot it until your bug report...
This will handle ep NFOs correctly.
Will have to correct other issues reported now, didn't have a look yet
- season-specials-poster.jpg is present for me on latest code, please report.
- Only export if Season_Art != Show_Art to avoid multiple copies of the same image
- Switch the preferences from lists to input so we can customize filenames.

**Comment by BrutuZ** [comment]
After this update all episodes have their corresponding NFO files and Season 00 (Specials) is parsed too.
Sorry for the delayed feedback.

**Comment by ZeroQI** [comment]
@BrutuZ excellent. You reported and made me fix a big issue and i made more progress in 2 days tan in 3 weeks. Thank you.
- Added test to avoid multiple copies between season and series for posters and fanart
- added multi-genre and collection nfo support (as tag)
- added local files listing in series nfo
- added collection items and local files
Please test and feedback. this should include all but 'Switch the preferences from lists to input so we can customize filenames.' as i included all xbmc compatible formats to my knowledge so i do not see why any further filename format is needed. If you have any technical reason, i am curious

**Comment by BrutuZ** [comment]
Just tested the latest commit (24f00df) and `SeasonXX-fanart` are still being copied as duplicates.
Could be also related to the NFO issue that expected a specific folder structure?

As for the filename customization, it's not so much for the filename itself (although I did add the `-poster` suffix on the DefaultPrefs file) as it is for customizing the destination folder.
For now I (and kodi) like to keep the images bundled in the series' root folder, but in the future would like to experiment other organization schemes, like a _.metadata_ folder

**Comment by ZeroQI** [comment]
i followed https://support.plex.tv/articles/200220717-local-media-assets-tv-shows/ for location
Coded to put the ep thumb/nfo with episode name, season media in season folder if exist otherwise series folder, so the import is done by local media assets agent. Seasonxx-poster.ext probably isn't loaded back by local media assets.
an option to move all in another folder if located at the root (_metadata) but need to make the agent work fully first.

Could I please have the logs for SeasonXX-fanart still being copied as duplicates ? need to see if filenames are identical or have to compare pictures...

**Comment by BrutuZ** [comment]
Fixing before adding more features that can turn into more points of failure. Sounds very reasonable 👍 

Filenames are different (`fanart.jpg` and `season01-fanart.jpg`), the image is the same and so is the Plex URL (`/library/metadata/245095/art/1537682488`). Compare that perhaps?

Just in case:
- https://pastebin.com/HuKMYtJk
- https://i.imgur.com/CTn51r3.png

**Comment by ZeroQI** [comment]
the season xml list only one when there is duplicate. It include the media id so checked its presence, easier...
Please test new version and feedback any issue remaining

**Comment by BrutuZ** [comment]
Duplicate images solved too on 63197b2. All reported issues fixed 👍
